src/util/BlankNodeManager.h

```diff
@@ -91,25 +92,41 @@ class BlankNodeManager {
     // BlankNodeIndex blocks.
     template <std::ranges::range R>
     void mergeWith(const R& localBlankNodeManagers) {
-      std::ranges::copy_if(
-          localBlankNodeManagers, std::back_inserter(otherManagers_),
-          [](const std::shared_ptr<const LocalBlankNodeManager>& l) -> bool {
-            return l != nullptr;
-          });
+      auto inserter = std::back_inserter(otherBlocks_);
+      for (const auto& l : localBlankNodeManagers) {
+        if (l == nullptr) {
+          continue;
```

Comment:
```suggestion
        if (l == nullptr || (l->blocks_->empty() && l->otherBlocks_->empty()) {
          continue;
```

Reply:
As small optimization for the very normal case that there are no blank nodes anywhere (Especially with the other comment that avoids the shared_ptr copies.

Reply:
`nullptr` already is the normal case, as a `LocalVocab` only inits its `localBlankNodeManager_` when either calling `getBlankNodeIndex` or merging other vocabs (containing at least one `localBlankNodeManager_`).